plugin/parser/differ/mysql/differ.go

```diff
@@ -0,0 +1,45 @@
+// Package mysql provides the MySQL differ plugin.
+package mysql
+
+import (
+	"bytes"
+
+	"github.com/pingcap/tidb/parser/ast"
+	"github.com/pingcap/tidb/parser/format"
+)
+
+// SchemaDiff returns the schema diff.
+func SchemaDiff(old, new []ast.StmtNode) (string, error) {
+	oldTableList := make([]*ast.CreateTableStmt, 0)
+	oldTableMap := make(map[string]int)
+	tableDiff := make([]ast.Node, 0)
+	for _, node := range old {
+		if stmt, ok := node.(*ast.CreateTableStmt); ok {
+			tableName := stmt.Table.Name.String()
+			oldTableMap[tableName] = len(oldTableList)
+			oldTableList = append(oldTableList, stmt)
+		}
+	}
+	for _, node := range new {
+		if stmt, ok := node.(*ast.CreateTableStmt); ok {
+			if _, ok := oldTableMap[stmt.Table.Name.String()]; !ok {
+				stmt.IfNotExists = true
```

Comment:
For SDL, no need to consider the `IF NOT EXISTS` option

Reply:
We can do our best to maintain idempotency. For example, the diff result is
```SQL
CREATE TABLE book(id INT);
CREATE TABLE author(id INT, name VARCHAR(50));
```
And I failed before running the second statement. If it has `IF NOT EXISTS` option, I can click the retry button without mental burden and not get another error because the book table already exists.